dietpi/dietpi-software

```diff
@@ -11615,6 +11655,62 @@ _EOF_
 			G_EXEC chown -R ffsync:ffsync /mnt/dietpi_userdata/firefox-sync
 
 		fi
+		
+                software_id=183 # Bitwarden RS
+                if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+                    Banner_Configuration
+
+                    Create_User -d /mnt/dietpi_userdata/bitwarden_rs bitwarden_rs
+
+                    # Make dir and data dir needed to run
+                    G_EXEC mkdir -p /mnt/dietpi_userdata/bitwarden_rs/data
+
+                    # Make and edit the template config to have no web vault
+                    G_EXEC cp /opt/bitwarden_rs/.env.template /mnt/dietpi_userdata/bitwarden_rs/bitwarden_rs.env
+                    G_EXEC sed -i 's/# WEB_VAULT_ENABLED=true/WEB_VAULT_ENABLED=false/g' /mnt/dietpi_userdata/bitwarden_rs/bitwarden_rs.env
+
+                    # Permissions
+                    G_EXEC chown -R bitwarden_rs:bitwarden_rs /mnt/dietpi_userdata/bitwarden_rs
+                    G_EXEC chmod +x /opt/bitwarden_rs/target/release/bitwarden_rs
+
+                    # Systemd service
+                    cat << _EOF_ > /etc/systemd/system/bitwarden_rs.service
+[Unit]
+Description=Bitwarden Server (Rust Edition)
+Documentation=https://github.com/dani-garcia/bitwarden_rs
+
+After=dietpi-boot.service network.target
+
+
+[Service]
+# The user/group bitwarden_rs is run under. the working directory (see below) should allow write and read access to this user/group
+User=bitwarden_rs
+Group=bitwarden_rs
+# The location of the .env file for configuration
+EnvironmentFile=/mnt/dietpi_userdata/bitwarden_rs/bitwarden_rs.env
+# The location of the compiled binary
+ExecStart=/opt/bitwarden_rs/target/release/bitwarden_rs
+# Set reasonable connection and process limits
+LimitNOFILE=1048576
+LimitNPROC=64
+# Isolate bitwarden_rs from the rest of the system
+PrivateTmp=true
+PrivateDevices=true
+ProtectHome=true
+ProtectSystem=strict
+# Only allow writes to the following directory and set it to the working directory (user and password data are stored here)
+WorkingDirectory=/mnt/dietpi_userdata/bitwarden_rs
+ReadWriteDirectories=/mnt/dietpi_userdata/bitwarden_rs
```

Comment:
Just to keep an eye on: I'm not sure if `PrivateTmp=true` implies that the service has write permissions to it's own isolated `/tmp` directory (would be reasonable) but it might be possible to add `/tmp` to `ReadWriteDirectories` as well for that.

Reply:
Again, I just copied from the bitwarden_rs example service

Reply:
Ah okay then this should work. A test will show quickly anyway 😉.